### ISSUE TITLE
Start date is off for 3 and 4 day view 

### ISSUE THREAD

**Issue by mikeyyyzhao** [kvyatkovskys/KVKCalendar#249]
**Describe the bug**
When I set the initial date (`CalendarView(frame: frame, date: Date(), style: style, years: 4)`), the calendar on 3 and 4 day view starts at the end of the date. For example, if today is April 11 and I'm in 3 day view, the 3 day view window is April 9, April 10, and April 11th. For most people on 3 day view, it's a rolling 3 day view so the window starts at April 11, April 12, and April 13. Maybe this isn't a bug but is it possible on initialize the have the initial date be the start of the rolling window?

To reproduce:
1. start with any of the example sample code
2. set CalendarView on init to 
```
var style = Style()
style.defaultType = .week
style.week.daysInOneWeek = 3
CalendarView(
            frame:frame,
            date: Date(),
            style: style
        )
```
3. the input date will not be in first date in the window

Note:
It looks like the number of days before `date` for CalendarView varies depending on `daysInOneWeek`

![Screen Shot 2022-04-11 at 2 51 00 PM](https://user-images.githubusercontent.com/4956032/162809153-a6fa3c37-36de-4605-aebc-6c416d82374a.png)

Example from Google Calendar
![IMG_013913825056-1](https://user-images.githubusercontent.com/4956032/162809262-8d79f96f-67d3-4a35-b29e-288bf98c9f41.jpeg)

**Comment by ErSahil56** [comment]
@mikeyyyzhao @kvyatkovskys Can you fix this bug please asap.

**Comment by kvyatkovskys** [comment]
You can use the code from this pull request https://github.com/kvyatkovskys/KVKCalendar/pull/250 until I get answers and approve these changes.

**Comment by ErSahil56** [comment]
@kvyatkovskys can you check issue https://github.com/kvyatkovskys/KVKCalendar/issues/251. Its quite important for me.

**Comment by mikeyyyzhao** [comment]
Resolved with #250
